src/App.tsx

```diff
@@ -1,204 +1,68 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useContext, useEffect, useState } from 'react';
 
-const USER_ID = 0;
+import { DispatchContext, StateContext } from './Store';
+
+import { getTodos } from './api/todos';
+import { filterTodos } from './services/filterTodos';
+
+import { USER_ID } from './lib/user';
+
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const dispatch = useContext(DispatchContext);
+  const { todos, filter } = useContext(StateContext);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
```

Comment:
try to rewrite your code without the separate state for filtered todos, it's redundant